.github/workflows/example.yml

```diff
@@ -27,9 +27,27 @@ permissions:
   contents: read
 
 jobs:
+  # Change detection in action golang code
+  changes:
+    runs-on: ubuntu-latest
+    # Required permissions
+    permissions:
+      pull-requests: read
+    outputs:
+      action: ${{ steps.filter.outputs.action }}
+    steps:
+      # For pull requests it's not necessary to checkout the code
+      - uses: dorny/paths-filter@v3
```

Comment:
What's the benefit of this instead of using the built-in path triggers i GitHub?

Reply:
This is to decide whether or not to use pre-compiled binary executable from releases, or if a freshly-compiled binary is needed to actually tests the changes.

For example if we make changes in `action/` golang code, they would not be tested if the executable is downloaded from latest release. In the case we have to compile on the fly.

Reply:
Yes, I got that. I was just wondering what the benefit of doing this via a thirdparty action is vs the built-in `paths` filter: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-your-pull_request-workflow-based-on-files-changed-in-a-pull-request

Reply:
How could we use that? It is a trigger.

We could use it but then we would have to copy-paste the workflow - one for downloading from releases and one for compilation.

Reply:
Ah so you want to make certain steps optional without having to split the yaml and potentially duplicate code. Got it.

Reply:
Yes, exatly. I have added `input` to the action called `compile` which switches between compilation mode and download mode. This is where I feed in the boolean output of the `dorny/paths-filter`.